pyccel/codegen/printing/ccode.py

```diff
@@ -822,13 +822,47 @@ def _print_PythonRound(self, expr):
     def _print_PythonMinMax(self, expr):
         arg = expr.args[0]
         primitive_type = arg.dtype.primitive_type
+        variadic_args = False
+        can_compare = True
 
-        if primitive_type is PrimitiveFloatingPointType() and len(arg) == 2:
+        if isinstance(primitive_type, (PrimitiveFloatingPointType, PrimitiveIntegerType)):
+            variadic_args = True
+
+        if primitive_type is PrimitiveComplexType():
+            can_compare = False
+
+        if isinstance(arg, Variable) and isinstance(arg.class_type , HomogeneousTupleType):
+            arg = PythonTuple(*arg)
+
+        if isinstance(arg, (PythonTuple, PythonList)) and variadic_args:
+            key = self.get_c_type(arg.class_type.element_type)
+            self.add_import(Import('stc/common', AsName(VariableTypeAnnotation(arg.dtype), key)))
+            return  f'{key}_{expr.name}({len(arg.args)}, {", ".join(self._print(a) for a in arg.args)})'
+        elif isinstance(arg, Variable):
+            if isinstance(arg.class_type, HomogeneousListType) and can_compare:
+                class_type = arg.class_type
+                c_type = self.get_c_type(class_type)
+                arg_obj = self._print(ObjectAddress(arg))
+                self.add_import(Import('stc/vec', AsName(VariableTypeAnnotation(class_type), c_type)))
+                return f'{c_type}_{expr.name}({arg_obj})'
+            elif isinstance(arg.class_type, HomogeneousSetType) and can_compare:
+                class_type = arg.class_type
+                c_type = self.get_c_type(class_type)
+                arg_obj = self._print(ObjectAddress(arg))
+                self.add_import(Import('stc/hset', AsName(VariableTypeAnnotation(class_type), c_type)))
+                return f'{c_type}_{expr.name}({arg_obj})'
```

Comment:
There is only 1 line difference between these 2. Maybe unify?
```suggestion
            if isinstance(arg.class_type, (HomogeneousListType, HomogeneousSetType)) and can_compare:
                class_type = arg.class_type
                c_type = self.get_c_type(class_type)
                arg_obj = self._print(ObjectAddress(arg))
                import_loc = 'stc/vec' if isinstance(arg.class_type, HomogeneousListType)) else 'stc/hset'
                self.add_import(Import('stc/vec', AsName(VariableTypeAnnotation(class_type), c_type)))
                return f'{c_type}_{expr.name}({arg_obj})'
```
This may be slightly slower though so I don't know if the trade-off is worth it.